yt_dlp/extractor/cbc.py

```diff
@@ -209,289 +185,243 @@ def _real_extract(self, url):
+            'series': 'Schitt\'s Creek',
+            'season': 'Season 6',
+            'season_number': 6,
+            'episode': 'Smoke Signals',
+            'episode_number': 1,
+            'episode_id': 'schitts-creek/s06e01',
         },
+        'params': {'format': 'bv'},  # No format has audio and video combined
+        'skip': 'Geo-restricted to Canada',
     }, {
         # geo-restricted to Canada, bypassable
-        'url': 'http://watch.cbc.ca/arthur/all/1ed4b385-cd84-49cf-95f0-80f004680057',
+        # This video requires an account in the browser, but works fine in yt-dlp
+        'url': 'https://gem.cbc.ca/media/schitts-creek/s01e01',
+        'md5': '297a9600f554f2258aed01514226a697',
         'info_dict': {
-            'id': '1ed4b385-cd84-49cf-95f0-80f004680057',
-            'title': 'Arthur',
-            'description': 'Arthur, the sweetest 8-year-old aardvark, and his pals solve all kinds of problems with humour, kindness and teamwork.',
+            'id': 'schitts-creek/s01e01',
+            'ext': 'mp4',
+            'title': 'The Cup Runneth Over',
+            'description': 'md5:9bca14ea49ab808097530eb05a29e797',
+            'thumbnail': 'https://images.radio-canada.ca/v1/synps-cbc/episode/perso/cbc_schitts_creek_season_01e01_thumbnail_v01.jpg?im=Resize=(Size)',
+            'series': 'Schitt\'s Creek',
+            'season_number': 1,
+            'season': 'Season 1',
+            'episode_number': 1,
+            'episode': 'The Cup Runneth Over',
+            'episode_id': 'schitts-creek/s01e01',
+            'duration': 1309,
+            'categories': ['comedy'],
         },
-        'playlist_mincount': 30,
+        'params': {'format': 'bv'},  # No format has audio and video combined
+        'skip': 'Geo-restricted to Canada',
     }, {
-        'url': 'https://gem.cbc.ca/media/this-hour-has-22-minutes/season-26/episode-20/38e815a-0108c6c6a42',
-        'only_matching': True,
+        # geo-restricted to Canada, bypassable
+        # TV show playlist, all public videos at time of coding (2021-09)
+        'url': 'https://gem.cbc.ca/media/schitts-creek/s06',
+        'playlist_count': 16,
+        'info_dict': {
+            'id': 'schitts-creek/s06',
+            'title': 'Season 6',
+            'description': 'md5:6a92104a56cbeb5818cc47884d4326a2',
+        },
+        'skip': 'Geo-restricted to Canada',
     }]
+    _API_BASE = 'https://services.radio-canada.ca/ott/cbc-api/v2/'
+    _API_VIDEO = _API_BASE + 'assets/'
+    _API_SHOW = _API_BASE + 'shows/'
+    _BROWSER_BASE = 'https://gem.cbc.ca/media/'
+    # Regexes for analyzing video_id - the id group of _VALID_URL
+    _VIDEO_ID = r'[0-9a-z-]+/s[0-9]+[a-z][0-9]+'
+    _PLAYLIST_ID = r'([0-9a-z-]+)/s([0-9]+)'
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        rss = self._call_api('web/browse/' + video_id, video_id)
-        return self._parse_rss_feed(rss)
-
+        url_id = self._match_id(url)
```

Comment:
Set this as `display_id` in the infodict

Reply:
This variable is the same as `playlist_id` or `show_id`, which is set to `id` in the infodict. So there's no need to set `display_id` as the same thing. Once I separate playlists into their own extractor this will be clearer and I won't rename the variable.